### PR TITLE
Add GCP cloud function processing

### PR DESCRIPTION
**Pull request by Sciguymjm**
Still WIP. We'll need to decide how to handle this, since the request will be blocking the worker for the entire processing time. I believe we should have two queues. One for GCP with a ton of workers, and our regular celery queue. We only add to GCP queue if the main celery queue size is greater than a set value. We'd put this logic in /api/parse.

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -1,10 +1,8 @@
 <component name="InspectionProjectProfileManager">
   <profile version="1.0">
     <option name="myName" value="Project Default" />
-    <inspection_tool class="PyArgumentEqualDefaultInspection" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
-    <inspection_tool class="PyAugmentAssignmentInspection" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
     <inspection_tool class="RequiredAttributes" enabled="true" level="WARNING" enabled_by_default="true">
-      <option name="myAdditionalRequiredHtmlAttributes" value="theme,classes,width,history,location,match" />
+      <option name="myAdditionalRequiredHtmlAttributes" value="theme,classes,width" />
```
Comment (dtracers) [inline .idea/inspectionProfiles/Project_Default.xml]: i feel like this may be out of date?

File: backend/blueprints/spa_api/spa_api.py [hunk 2]
```diff
@@ -28,7 +29,6 @@
 from .service_layers.player.player_profile_stats import PlayerProfileStats
 from .service_layers.player.player_ranks import PlayerRanks
 from .service_layers.queue_status import QueueStatus
-from .service_layers.replay.basic_stats import BasicStatChartData
```
Comment (dtracers) [inline backend/blueprints/spa_api/spa_api.py]: why was this file touched

File: webapp/package.json [hunk 3]
```diff
@@ -7,11 +7,10 @@
     "@fortawesome/free-brands-svg-icons": "^5.3.1",
     "@fortawesome/free-solid-svg-icons": "^5.3.1",
     "@fortawesome/react-fontawesome": "^0.1.3",
-    "@material-ui/core": "^3.1.2",
+    "@material-ui/core": "^3.0.3",
```
Comment (dtracers) [inline webapp/package.json]: why was this file touched?

File: .idea/inspectionProfiles/Project_Default.xml (hunk 1 above)
Comment (twobackfromtheend) [reply .idea/inspectionProfiles/Project_Default.xml]: use the red one, not the green one. bad merge.

File: webapp/package.json (hunk 3 above)
Comment (Sciguymjm) [reply webapp/package.json]: I'm not sure why. But I can revert those changes.

File: backend/tasks/celery_tasks.py [hunk 4]
```diff
@@ -151,6 +154,20 @@ def parse_replay_gcp(self, fn):
...
 
+@celery.task(base=DBTask, bind=True, priority=9)
+def parse_replay_gcp(self, fn):
+    with open(fn, 'rb') as f:
+        encoded_file = base64.b64encode(f.read())
+    r = requests.post(GCP_URL, data=encoded_file)
+    response = r.json()
+    i = io.BytesIO(base64.b64decode(response['proto']))
+    protobuf_game = ProtobufManager.read_proto_out_from_file(i)
+    i2 = io.BytesIO(base64.b64decode(response['pandas']))
```
Comment (twobackfromtheend) [inline backend/tasks/celery_tasks.py]: pls.

I feel `r` does not need to be created - just chain `.json()` on.

`i` and `i2` can definitely do with better names.

File: backend/blueprints/spa_api/spa_api.py (hunk 2 above)
Comment (Sciguymjm) [reply backend/blueprints/spa_api/spa_api.py]: This is required, BasicStatChartData does not exist anymore.

File: webapp/package.json (hunk 3 above)
Comment (twobackfromtheend) [reply webapp/package.json]: yeah please revert the changes in this file.

File: backend/tasks/celery_tasks.py (hunk 4 above)
Comment (Sciguymjm) [reply backend/tasks/celery_tasks.py]: Old code, I swear I've changed ;)

File: webapp/tslint.json [hunk 5]
```diff
@@ -52,6 +52,7 @@
     "quotemark": [true, "double", "avoid-escape", "jsx-double"],
     "space-before-function-paren": [true, "never"],
     "whitespace": true,
+    "member-ordering": false,
```
Comment (twobackfromtheend) [inline webapp/tslint.json]: and this.

File: backend/blueprints/spa_api/spa_api.py (hunk 2 above)
Comment (twobackfromtheend) [reply backend/blueprints/spa_api/spa_api.py]: but why's PandasManager added as an import.

Comment (Sciguymjm) [reply backend/blueprints/spa_api/spa_api.py]: o

Comment (Sciguymjm) [reply backend/blueprints/spa_api/spa_api.py]: It was used in the old proto processing code (to verify we decoded correctly) but now its unused.

**Pull request closed without merging**